seed/analysis_pipelines/bsyncr.py

```diff
@@ -108,74 +109,62 @@ def _prepare_all_properties(analysis_property_view_ids, analysis_id, progress_da
-                    debug_message=''
+                    user_message=f'Error preparing bsyncr input: {error}',
+                    debug_message='',
                 )
-                continue
-
-            # arbitrarily choosing the first meter for now
-            meter = meters[0]
-
-            bsync_doc, errors = _build_bsyncr_input(analysis_property_view, meter)
-            if errors:
-                for error in errors:
-                    AnalysisMessage.log_and_create(
-                        logger=logger,
-                        type_=AnalysisMessage.ERROR,
-                        analysis_id=analysis.id,
-                        analysis_property_view_id=analysis_property_view.id,
-                        user_message=f'Error preparing bsyncr input: {error}',
-                        debug_message='',
-                    )
-                continue
+            continue
 
-            analysis_input_file = AnalysisInputFile(
-                content_type=AnalysisInputFile.BUILDINGSYNC,
-                analysis=analysis
-            )
-            analysis_input_file.file.save(f'{analysis_property_view.id}.xml', ContentFile(bsync_doc))
-            analysis_input_file.clean()
-            analysis_input_file.save()
-            input_file_paths.append(analysis_input_file.file.path)
-        except Exception as e:
-            AnalysisMessage.log_and_create(
-                logger=logger,
-                type_=AnalysisMessage.ERROR,
-                analysis_id=analysis.id,
-                analysis_property_view_id=analysis_property_view.id,
-                user_message='Unexpected error occurred while preparing input for property',
-                debug_message='',
-                exception=e,
-            )
-            pass
+        analysis_input_file = AnalysisInputFile(
+            content_type=AnalysisInputFile.BUILDINGSYNC,
+            analysis=analysis
+        )
+        analysis_input_file.file.save(f'{analysis_property_view.id}.xml', ContentFile(bsync_doc))
+        analysis_input_file.clean()
+        analysis_input_file.save()
+        input_file_paths.append(analysis_input_file.file.path)
 
     if len(input_file_paths) == 0:
         pipeline = BsyncrPipeline(analysis.id)
         message = 'No files were able to be prepared for the analysis'
-        pipeline.fail(message, progress_data_key=progress_data.key, logger=logger)
+        pipeline.fail(message, logger, progress_data_key=progress_data.key)
         # stop the task chain
-        raise AnalysisPipelineException(message)
+        raise StopAnalysisTaskChain(message)
 
 
-@shared_task
-@check_analysis_status(Analysis.CREATING)
-def _finish_preparation(analysis_id, progress_data_key):
+@shared_task(bind=True)
+@analysis_pipeline_task(Analysis.CREATING)
+def _finish_preparation(self, analysis_id, progress_data_key):
```

Comment:
note that adding `bind=True` gives us access to the celery task instance in the function (self)